src/x3f_io.c

```diff
@@ -78,6 +78,12 @@ static int x3f_get4(FILE *f)
 #define GET1(_v) do {(_v) = x3f_get1(I->input.file);} while (0)
 #define GET2(_v) do {(_v) = x3f_get2(I->input.file);} while (0)
 #define GET4(_v) do {(_v) = x3f_get4(I->input.file);} while (0)
+#define GET4F(_v)				\
+  do {						\
+    union {int32_t i; float f;} _tmp;		\
+    _tmp.i = x3f_get4(I->input.file);		\
+    (_v) = _tmp.f;				\
+  } while (0)
```

Comment:
This is a commonly used trick for making multi-statement macros. Not
very pretty and kind of weird, but it gets the job done.

On mar, 2015-07-28 at 08:13 -0700, mmroden wrote:
> In src/x3f_io.c:
> 
> > @@ -78,6 +78,12 @@ static int x3f_get4(FILE *f)
> >  #define GET1(_v) do {(_v) = x3f_get1(I->input.file);} while (0)
> >  #define GET2(_v) do {(_v) = x3f_get2(I->input.file);} while (0)
> >  #define GET4(_v) do {(_v) = x3f_get4(I->input.file);} while (0)
> > +#define GET4F(_v)				\
> > +  do {						\
> > +    union {int32_t i; float f;} _tmp;		\
> > +    _tmp.i = x3f_get4(I->input.file);		\
> > +    (_v) = _tmp.f;				\
> > +  } while (0)
> 
> (potentially stupid) question-- why the use of the do/while loop, when
> the while is set to 0? Why not just call the routine directly, with no
> do-while? As I'm reading this, this code just calls the do-while
> interior once then exits, so not sure what do-while gets you here.
> 
> —
> Reply to this email directly or view it on GitHub.
> 
>